modules/samples/virtual_firealarm/src/org.wso2.carbon.device.mgt.iot.sample.virtual.firealarm.service.impl/src/main/java/org/wso2/carbon/device/mgt/iot/sample/virtual/firealarm/service/impl/VirtualFireAlarmService.java

```diff
@@ -645,20 +649,16 @@ public String requestTemperature(@HeaderParam("owner") String owner,
 					break;
 
 				default:
-					replyMsg = "Requested protocol '" + protocolString + "' is not supported";
-					response.setStatus(Response.Status.NOT_ACCEPTABLE.getStatusCode());
-					return replyMsg;
-			}
+                    response.setStatus(Response.Status.NOT_ACCEPTABLE.getStatusCode());
+            }
+            temperatureRecord = DataHolder.getThisInstance().getTemperature(deviceId);
 		} catch (DeviceManagementException e) {
-			replyMsg = e.getErrorMessage();
 			response.setStatus(Response.Status.INTERNAL_SERVER_ERROR.getStatusCode());
```

Comment:
put appropriate error log messages as a way of conveying an error has occurred at server-side.